### ISSUE TITLE
[13.1.2] Freezes "Creating an optimized production build"

### ISSUE THREAD

**Issue by Trallp** [vercel/next.js#45005]
### Verify canary release

- [X] I verified that the issue exists in the latest Next.js canary release

### Provide environment information

```
 Operating System:
      Platform: darwin
      Arch: x64
      Version: Darwin Kernel Version 22.2.0: Fri Oct 28 06:51:16 PDT 2022; root:xnu-8792.60.51~15/RELEASE_ARM64_T6000
    Binaries:
      Node: 14.20.0
      npm: 6.14.17
      Yarn: 3.2.1
      pnpm: 6.0.1
    Relevant packages:
      next: 13.1.3-canary.4
      eslint-config-next: 13.1.3-canary.4
      react: 18.2.0
      react-dom: 18.2.0
```

### Link to the code that reproduces this issue

https://github.com/Trallp/next-13.1.2-build-freeze-repro

### To Reproduce

yarn build

### Describe the Bug

The build is freeze if you turn off swcMinify and turn on productionBrowserSourceMaps

next,config.js

`module.exports = {
  swcMinify: false,
  productionBrowserSourceMaps: true,
}`

### Expected Behavior

Build is successful

**Comment by ibraelillo** [comment]
I'm also having this bad behavior after upgrading to 13.1.2.

**Comment by jayan-blutui** [comment]
I'm facing the same issue. I have been pulling my hair trying to figure out the root of the issue. 👨🏽‍🦲

Turns out the issue was happening on local and Vercel deployments. I upgraded to a `canary` version and local worked just fine, I deployed the `canary` version to Vercel and it fails there. I also tried changing the Node versions in Vercel, that did not help.

**Comment by jayan-blutui** [comment]
Update: I managed to solve this by removing majority of the pages in the `pages` directory. I then committed that to trigger a deployment in Vercel, which successfully built. I then re-added the removed pages and committed again to trigger the "actual" build, which also successfully built.

This is very strange and I have a feeling it has to do with some sort of cache.

**Comment by Trallp** [comment]
Yes, the build freezes for various reasons, as if even without the conflict of parameters, which I described in the issue. But it's just a case which I found to reproduce the problem in the clean and new repo

**Comment by ibraelillo** [comment]
I managed to remove the build issue by removing the image optimization configuration

**Comment by jasondecastro** [comment]
I resolved this by rebuilding my `yarn.lock` file.

**Comment by KakakuCZ** [comment]
I have same issue

**Comment by Trallp** [comment]
On 13.1.6, the project build is still not going to. We have a project of 1500 pages, and even with the disabled sourcemap it still freezes

**Comment by Trallp** [comment]
Fixed for me by [v13.1.7-canary.0](https://github.com/vercel/next.js/releases/tag/v13.1.7-canary.0)

**Comment by AleksandrKarashev** [comment]
same
